src/transformStateWithClones.js

```diff
@@ -7,7 +7,26 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let currentState = { ...state };
+  const stateHistory = [];
+
+  for (const action of actions) {
```

Comment:
Use destructuring for `action`
```suggestion
  for (const action of actions) {
  const { type, ..... } = action;
```